extern/noah-owp-modular/CMakeLists.txt

```
@@ -1,105 +1,66 @@
-
-#Use ../../../FindNetCDF.cmake, which looks for config package, pkgconfig, and standard file locations
-#in that order
-###### 
-# Updates to noah-owp-modular allow a build of the bmi module that does not require netcdf
-# The NGEN_OUTPUT_ACTIVE compiler directive disables all netcdf related code, hence no need
-# to link the netcdf library. If at some point this becomes required, the following section
-# can be uncommented to provide mechanics to look for and/or define netcdf requirements.
-# Also, the target link library line will need uncommeted below.
-######
-###### NETCDF ######
-# find_package (NetCDF COMPONENTS F90)
-# set(netCDF_INCLUDE_DIR ${NetCDF_INCLUDE_DIRS} CACHE PATH "NetCDF Include Directory" )
-# set(netCDF_C_LIB ${NetCDF_LIBRARIES} CACHE PATH 
-# "The path to the C shared object library file, libnetcdf.so or the library name if it can be found by the linker, i.e. netcdf" )
-# if(NetCDF_FOUND)
-#   #This provides a little diagnostic output, but most importantly, gives cache varabiles
-#   #for each piece of the netcdf puzzle that a user might need to directly override
-#   #if netcdf isn't configured correctly or cannot be found using the known methods.
-#   message("Found netCDF package")
-#   message("NetCDF inc: " ${NetCDF_INCLUDE_DIR})
-#   message("NetCDF lib: " ${NetCDF_LIBRARIES})
-# else()
-#   message(SEND_ERROR "Could not find NetCDF, install the depdency or\n 1) Set the NetCDF_DIR to a source build\n 2) set the netCDF* cache variables as appropriate")
-# endif()
-# # Ensure we can find the netcdf.mod file, leave a cache var that can be set if we can't
-# find_path(_netCDF_MOD_PATH "netcdf.mod" ${netCDF_INCLUDE_DIR} REQUIRED NO_CACHE)
-# set(netCDF_MOD_PATH ${_netCDF_MOD_PATH} CACHE PATH "Directory containing the netcdf.mod fortran module")
+# NGen NOAH-OWP-MODULAR listfile shim
 #
-# # For some reason the netcdf config.cmake doesn't set the netcdff lib correctly in
-# # ${netCDF_Libararies}, so it only links the C lib (netcdf) and not the fortran lib (netcdff)
-# # So find that library explicitly and make sure we link with it.
-# # also, the FindNetCDF.cmake doesn't seem to do it correctly either.
-# find_library(_netCDF_FORTRAN_LIB netcdff REQUIRED  NO_CACHE)
-# set(netCDF_FORTRAN_LIB ${_netCDF_FORTRAN_LIB} CACHE PATH 
-# "The path to the C shared object library file, libnetcdff.so or the library name if it can be found by the linker, i.e. netcdff")
-##### END NETCDF #####
-
-set(MODEL_SOURCE_DIR noah-owp-modular/src/)
-set(BMI_SOURCE_DIR noah-owp-modular/bmi/)
-
-file(GLOB BMI_SOURCE ${BMI_SOURCE_DIR}*.f90)
-file( GLOB MODEL_SOURCES ${MODEL_SOURCE_DIR}*.f90)
-
-list(APPEND MODEL_SOURCES noah-owp-modular/driver/AsciiReadModule.f90
-                          noah-owp-modular/driver/OutputModule.f90)
-
-if(WIN32)
-    add_library(surfacebmi ${BMI_SOURCE} )
+# Requires:
+# - NGEN_WITH_BMI_FORTRAN=ON
+#
+# Output Targets:
+# - `surfacebmi`
+
+project(noah-owp-modular VERSION 1.0.0 LANGUAGES Fortran)
```

Should we do something like this to avoid a potential collision with a future CMake build of noah?
```suggestion
project(noah-owp-modular-ngen-shim VERSION 1.0.0 LANGUAGES Fortran)
```

---

I'm not too worried about it, since we are only adding the parent subdirectory -- any listfiles in the submodule aren't being added. To that note though, these shims should probably be PR'd into their respective repositories

---

Ok, that's fine.